src/cdk-lib/bedrock/guardrails/guardrails.ts

```diff
@@ -0,0 +1,552 @@
+      resourceArns: [this.guardrailArn],
+      actions: ['bedrock:ApplyGuardrail'],
+      scope: this,
+    });
+  }
+}
+
+/******************************************************************************
+ *                        PROPS FOR NEW CONSTRUCT
+ *****************************************************************************/
+/**
+ * Properties for creating a Guardrail.
+ */
+export interface GuardrailProps {
+  /**
+   * The name of the guardrail.
+   */
+  readonly name: string;
+  /**
+   * The description of the guardrail.
+   */
+  readonly description?: string;
+  /**
+   * The message to return when the guardrail blocks a prompt.
+   *
+   * @default "Sorry, your query violates our usage policy."
+   */
+  readonly blockedInputMessaging?: string;
+  /**
+   * The message to return when the guardrail blocks a model response.
+   *
+   * @default "Sorry, I am unable to answer your question because of our usage policy."
+   */
+  readonly blockedOutputsMessaging?: string;
+  /**
+   * A custom KMS key to use for encrypting data.
+   *
+   * @default "Your data is encrypted by default with a key that AWS owns and manages for you."
+   */
+  readonly kmsKey?: IKey;
+  /**
+   * The content filters to apply to the guardrail.
+   */
+  readonly contentFilters?: filters.ContentFilter[];
+  /**
+   * Up to 30 denied topics to block user inputs or model responses associated with the topic.
+   */
+  readonly deniedTopics?: filters.DeniedTopic[];
+  /**
+   * The word filters to apply to the guardrail.
+   */
+  readonly wordFilters?: string[];
+  /**
+   * The managed word filters to apply to the guardrail.
+   */
+  readonly managedWordListFilters?: filters.ManagedWordFilterType[];
+  /**
+   * The PII filters to apply to the guardrail.
+   */
+  readonly piiFilters?: filters.PIIFilter[];
+  /**
+   * The regular expression (regex) filters to apply to the guardrail.
+   */
+  readonly regexFilters?: filters.RegexFilter[];
+  /**
+   * The contextual grounding filters to apply to the guardrail.
+   */
+  readonly contextualGroundingFilters?: filters.ContextualGroundingFilter[];
+}
+
+/******************************************************************************
+ *                        NEW CONSTRUCT DEFINITION
+ *****************************************************************************/
+/**
+ * Class to create a Guardrail with CDK.
+ */
+export class Guardrail extends GuardrailBase {
+  /**
+   * Import a guardrail given an ARN.
+   */
+  public static fromGuardrailArn(scope: Construct, id: string, guardrailArn: string): IGuardrail {
+    class Import extends GuardrailBase {
+      public readonly guardrailArn = guardrailArn;
+      public readonly guardrailId = Arn.split(guardrailArn, ArnFormat.SLASH_RESOURCE_NAME).resourceName!;
+    }
+
+    return new Import(scope, id);
+  }
+  /**
+   * Import a guardrail given an ID.
+   */
+  public static fromGuardrailId(scope: Construct, id: string, guardrailId: string): IGuardrail {
```

Comment:
See my other comment here about the ```fromGuardrailAttributes``` suggestion:  https://github.com/awslabs/generative-ai-cdk-constructs/pull/751#discussion_r1803790094

Maybe we could also add a method to import a Guardrail created from the CfnGuardrail L1 construct, something like:

```public static fromCfnGuardrail(cfnGuardrail: cfnGuardrail): IGuardrail```

It would make it easier for users who already work with the L1 construct to transition to this library, and coupled with the comment mentioned above, would help to apply the guardrails to agents.

As for the other comment, this might be part of a subsequent PR.

Reply:
Will do this in a later PR